### DISCUSSION TITLE
CreateOrderException more details

### DISCUSSION THREAD

**Discussion by Ivan-Melnyk** [SalesforceCommerceCloud/sfcc-ci#425] [Team Posts]
Hello All,

Does anyone of you know how to get more info about APIException type 'CreateOrderException' in case if OrderMgr.createOrder(Basket) did not work? I can see only 'dw.order.CreateOrderException: Order creation not possible for this basket.', is there any way how to get exact error (i.e. Inventory, Taxation, N/A price in Basket, Order replacement, etc.)?

Thanks,
Ivan.

**Comment by mdzafar** [comment]
createOrder method throws an APIException with type 'CreateOrderException' if any of the following conditions are encountered:

- any of the totals (net, gross, tax) of the basket is N/A
- any of the product items is not available (this takes previously reserved items into account)
- any campaign-based coupon in the basket is invalid (see CouponLineItem.isValid()
- the basket represents an order being edited, but the order has been already been replaced by another order
- the basket represents an order being edited, but the customer associated with the original order is not the same as the current customer

```
try
 {
   var order  = OrderMgr.createOrder(basket);
 }
 catch (e if e instanceof APIException && e.type === 'CreateOrderException')
 {
   // handle e
 }
```

**Comment by taras-holub** [comment]
Also take a look if all your totals are available. It may looks fine in debugger as the value is 0, but being not available as a price due to invalid currency etc.

**Comment by Ivan-Melnyk** [comment]
> createOrder method throws an APIException with type 'CreateOrderException' if any of the following conditions are encountered:
> 
> any of the totals (net, gross, tax) of the basket is N/A
> any of the product items is not available (this takes previously reserved items into account)
> any campaign-based coupon in the basket is invalid (see CouponLineItem.isValid()
> the basket represents an order being edited, but the order has been already been replaced by another order
> the basket represents an order being edited, but the customer associated with the original order is not the same as the current customer



> Also take a look if all your totals are available. It may looks fine in debugger as the value is 0, but being not available as a price due to invalid currency etc.

Guys, I know why it can happen, I also can read the official documentation :), but I want to put a bit more details in error log, not just "this Order can not be created".

**Comment by pravngaur** [comment]
Hi Ivan,

I am not with laptop, could not check but see
CreateOrderException class inherits the APIException class, which has the
properties which does exactly what you are looking for.

Please check and let us know if it works for yoy

**Comment by Ivan-Melnyk** [comment]
> Hi Ivan,
> 
> I am not with laptop, could not check but see
> CreateOrderException class inherits the APIException class, which has the
> properties which does exactly what you are looking for.
> 
> Please check and let us know if it works for yoy
> …

Hi Pravngaur, APIException does not provide anything extra than in dw.order.CreateOrderException.

**Comment by jayashree-raja** [comment]
Hi Ivan, We recently faced this issue and while investigating,the basket object was not created. We checked the taxation type that has to be included in invoice and then the shipping method.After correcting these two we were able to create  a successful order. Not sure if this will be useful for you we faced same exception and we were able to correct these.

**Comment by Ivan-Melnyk** [comment]
> Hi Ivan, We recently faced this issue and while investigating,the basket object was not created. We checked the taxation type that has to be included in invoice and then the shipping method.After correcting these two we were able to create a successful order. Not sure if this will be useful for you we faced same exception and we were able to correct these.

Once again, guys, I'm not asking about the reason of this error, I'm asking about how to get more details from SFCC API objects/classes.
